modules/apps/headless/headless-object/headless-object-test/src/testIntegration/java/com/liferay/headless/object/resource/v1_0/test/CollaboratorResourceTest.java

```diff
@@ -6,14 +6,572 @@
+					testGroup.getGroupId(), TestPropsValues.getUserId())));
+	}
+
+	private User _getUser() throws Exception {
+		User user = UserTestUtil.addUser();
+
+		_users.add(user);
+
+		return user;
+	}
+
+	private Collaborator _getUserCollaborator() throws Exception {
+		User user = _getUser();
+
+		return new Collaborator() {
+			{
+				actionIds = new String[] {
+					SharingEntryAction.VIEW.getActionId()
+				};
+				dateExpired = _randomDatePlusAYear();
+				externalReferenceCode = user.getExternalReferenceCode();
+				name = user.getFullName();
+				share = true;
+				type = "user";
+			}
+		};
+	}
+
+	private UserGroup _getUserGroup() throws Exception {
+		UserGroup userGroup = UserGroupTestUtil.addUserGroup();
+
+		_userGroups.add(userGroup);
+
+		return userGroup;
+	}
+
+	private Collaborator _getUserGroupCollaborator() throws Exception {
+		UserGroup userGroup = _getUserGroup();
+
+		return new Collaborator() {
+			{
+				actionIds = new String[] {
+					SharingEntryAction.VIEW.getActionId()
+				};
+				dateExpired = _randomDatePlusAYear();
+				externalReferenceCode = userGroup.getExternalReferenceCode();
+				name = userGroup.getName();
+				share = true;
+				type = "userGroup";
+			}
+		};
+	}
+
+	private Date _randomDatePlusAYear() {
+		Calendar calendar = CalendarFactoryUtil.getCalendar();
+
+		calendar.add(Calendar.YEAR, 1);
+		calendar.set(Calendar.SECOND, 0);
+		calendar.set(Calendar.MILLISECOND, 0);
+
+		return calendar.getTime();
+	}
+
+	private Collaborator _toCollaborator(SharingEntry sharingEntry)
+		throws Exception {
+
+		if (sharingEntry.getToUserId() > 0) {
+			User user = _userLocalService.getUser(sharingEntry.getToUserId());
+
+			return new Collaborator() {
+				{
+					actionIds = TransformUtil.transformToArray(
+						SharingEntryAction.getSharingEntryActions(
+							sharingEntry.getActionIds()),
+						SharingEntryAction::getActionId, String.class);
+					dateExpired = sharingEntry.getExpirationDate();
+					externalReferenceCode = user.getExternalReferenceCode();
+					name = user.getFullName();
+					share = sharingEntry.isShareable();
+					type = "user";
+				}
+			};
+		}
+
+		UserGroup userGroup = _userGroupLocalService.getUserGroup(
+			sharingEntry.getToUserGroupId());
+
+		return new Collaborator() {
+			{
+				actionIds = TransformUtil.transformToArray(
+					SharingEntryAction.getSharingEntryActions(
+						sharingEntry.getActionIds()),
+					SharingEntryAction::getActionId, String.class);
+				dateExpired = sharingEntry.getExpirationDate();
+				externalReferenceCode = userGroup.getExternalReferenceCode();
+				name = userGroup.getName();
+				share = sharingEntry.isShareable();
+				type = "userGroup";
+			}
+		};
+	}
+
+	private static ObjectDefinition _objectDefinition;
+
+	@Inject
+	private static ObjectDefinitionLocalService _objectDefinitionLocalService;
+
+	@Inject
+	private static ObjectFieldLocalService _objectFieldLocalService;
+
+	private static List<UserGroup> _userGroups;
```

Comment:
@Mikellorza, from memory I think that `@DeleteAfterTestRun` can also be applied to lists or arrays

Reply:
Testing it

Reply:
Yes, seems that it works